### DISCUSSION TITLE
Nemo in production. Audio conversion to mp3

### DISCUSSION THREAD

**Discussion by dtemnov** [NVIDIA-NeMo/NeMo#3916] [Q&A] [answered]
Hi folks,

My question is not related to Nemo in particular but rather to the conversion to mp3 in prod.
We are running Nemo container 22.01 and inference is working fine as well as .wav generation via Soundfile but at the next step we are converting the audio to .mp3 and then send it to backend.

So the issue is - after around 16000 successful conversions via pydub(AudioSegment.from_file(audio_path, format="wav").export(mp3_path, format="mp3")) the ffmpeg suddenly starts to fail, the same happens with the Lame(used via  subprocess.call(cmd, shell=True)).

<img width="941" alt="image (3)" src="https://user-images.githubusercontent.com/94976736/161089574-a8682747-15c6-4996-ae96-3a14db085cd4.png">

I have no idea how to debug this behavior so starting this topic to find if possibly someone had the similar experience and any solution to this issue(apart from restarting the container, obviously 😄)

**Comment by titu1994** [comment, marked as answer]
This seems to be something external to NeMo is causing your container to crash - ffmpeg related perhaps. We would instead request use of RIVA for much more efficient inference https://developer.nvidia.com/riva

**Reply by dtemnov** [reply]
I also think this is either ffmpeg or python(subprocess calls) related issue.  Thanks for you advice, I'm already exploring Riva for deployment.
Could you elaborate a lit bit about the way Riva does audio processing? I'm afraid that ffmpeg is also being used on the background and Riva won't solve the issue.

**Reply by titu1994** [reply]
Hmm I'm not familiar with the details of the implementation but I have not heard of issues as in this case when triton server is being used for inference so I would assume it works well enough without crashing

**Reply by dtemnov** [reply]
Okay, I'll look into Riva implemetation because the issue appears at the last step during the conversion to mp3, the container itself and wave generation keeps working fine... Anyway, thanks for your assistance.
